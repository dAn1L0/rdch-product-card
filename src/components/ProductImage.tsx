import React, { CSSProperties, useContext } from 'react'
import { ProductContext } from './ProductCard'

import noImage from '../assets/no-image.jpg'
import styles from '../styles/styles.module.css'

export interface Props {
  img?: string
  className?: string
  style?: CSSProperties
}

export const ProductImage = ({img, className, style}: Props) => {

  const { product } = useContext(ProductContext)

  let imgToShow: string
  let titleToShow: string

  if (img) {
    imgToShow = img
  } else if ( product.img ) {
    imgToShow = product.img
  } else {
    imgToShow = noImage
  }

  if ( product?.title ) {
    titleToShow = product?.title.toLowerCase().replace(' ','-')
  } else {
    titleToShow = 'product-image'
  }

  return (
    <img 
      className={ `${styles.productImg} ${className}` } 
      src={ imgToShow } 
      alt={ titleToShow }
      style={style}
    />
  )
}