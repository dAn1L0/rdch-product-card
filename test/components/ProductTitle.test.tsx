import * as React from 'react';
import renderer from 'react-test-renderer'
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('Pruebas en el <ProductTitle/>', () => {

  test('debe mostrar el componente correctamente con el tÃ­tulo personalizado', () => { 
    const wrapper = renderer.create(
      <ProductTitle
        title='Custom Product'
      />
    )

    expect( wrapper.toJSON() ).toMatchSnapshot();
    
  })

  test('debe mostrar el componente con el "title" del producto', () => {
    const wrapper = renderer.create(
      <ProductCard
        product={product1}
      >
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
    

  })

})