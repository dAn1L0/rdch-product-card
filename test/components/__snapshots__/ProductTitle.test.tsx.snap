// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <ProductTitle/> debe mostrar el componente con el "title" del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug
  </span>
</div>
`;

exports[`Pruebas en el <ProductTitle/> debe mostrar el componente correctamente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
